wof/wofpy_config.py

```diff
@@ -16,62 +16,82 @@
 
 Usage:
     wofpy_config INDIR
-    wofpy_config INDIR [--mode=<development,production>]
+    wofpy_config INDIR [--mode=<development,production>] [--overwrite=<soft,hard,none>]
 
-    wofpy_config (-h | --help | -v | --version)
+    wofpy_config (-h | --help | -v | --version | -m | --mode | -o | --overwrite)
 
 Examples:
     wofpy_config wofpyserver
+    wofpy_config wofpyserver --mode=production --overwrite=soft
 
 Arguments:
-  directory     Configuration directory.
+  directory               Configuration directory.
 
 Options:
-  -h --help     Show this screen.
-  -v --version  Show version.
-  --mode=development   deploy mode [default: development]
+  -h --help               Show this screen.
+  -v --version            Show version.
+  -m --mode=development   Deploympent mode [default: development].
+  -o --overwrite=none     Overwrite everything *hard*,
```

Comment:
The `click` vs `docopt` discussion is almost the same as `argparse` vs `optparse`.
a POSIX only parser seems limited but it is actually quite powerful and field tested, after all virtually all of Unix CLIs as like that.

Both `click` and `argarse` will allow you to create fancier CLI, with complex options and defaults, but the real question is: should you be creating a CLI like that :stuck_out_tongue_winking_eye:

On the other hand `click` does support subcommands while `docopt` would rely on pipes to get a similar functionality.

TL;DR it is a matter of taste but I like to be on the side of simplicity and easy manitainability.
So unless I am building a CLI tool that will have many subcommands, like `conda (-build,-env,-install,-create,etc)`, I prefer `optparse`, or its easier to use cousin `docopt`.